### PR TITLE
Avoid assuming RSG in non-C# projects

### PR DESCRIPTION
**Pull request by pranavkm**
* [x] While it's generally no-op to define it, we can be correct and only configure UseRazorSourceGenerator for C# projects where they're supported.

### CODE REVIEW

**Comment by adrianwright109** [comment]
Looks like this also fixes https://github.com/dotnet/aspnetcore/issues/33848

File: src/RazorSdk/Targets/Sdk.Razor.CurrentVersion.targets [hunk 1]
```diff
@@ -49,9 +49,7 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_TargetingNETCoreApp30OrLater>true</_TargetingNETCoreApp30OrLater>
         <_TargetingNET50OrLater>true</_TargetingNET50OrLater>
         <_TargetingNET60OrLater>true</_TargetingNET60OrLater>
-        <UseRazorSourceGenerator Condition="'$(UseRazorSourceGenerator)' == '' ">true</UseRazorSourceGenerator>
-        <!-- Required to support incremental source generator -->
-        <LangVersion Condition="'$(UseRazorSourceGenerator)' != '' ">preview</LangVersion>
+        <UseRazorSourceGenerator Condition="'$(Language)' == 'C#' AND '$(UseRazorSourceGenerator)' == '' ">true</UseRazorSourceGenerator>
```
Comment (pranavkm) [inline src/RazorSdk/Targets/Sdk.Razor.CurrentVersion.targets:52]:
```suggestion
        <!-- Required to support incremental source generator -->
        <LangVersion Condition="'$(UseRazorSourceGenerator)' != '' ">preview</LangVersion>
        <UseRazorSourceGenerator Condition="'$(Language)' == 'C#' AND '$(UseRazorSourceGenerator)' == '' ">true</UseRazorSourceGenerator>
```

**Comment by pranavkm** [comment]
@adrianwright109 unfortunately not. The compiler still seems to require a LangVersion=preview for incremental source generators.

@captainsafia could I get a quick review? I'd like to try and get this ported to rc1.

**Review by captainsafia — APPROVED** [review]

**Pull request merged**